pkg/util/error.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"errors"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+)
+
+func IsNotFound(err error) bool {
```

Comment:
To reviewer:
I unified `k8serrors.IsNotFound` with this customized function, based on the workaround solution mentioned in this open issue  https://github.com/kubernetes-sigs/controller-runtime/issues/2354 thanks @LeelaChacha bring this into attention